code/game/machinery/computer/bhangmeter.dm

```diff
@@ -0,0 +1,335 @@
+		</head>
+		<body>
+		<h2 style="text-align:center">Bhangmeter Database ([thisZ.name])</h2>
+		<table>
+		<tr>
+		<th style="width:0.2%">Time</th>
+		<th style="width:1%">Area</th>
+		<th style="width:0.2%">Epicenter</th>
+		<th style="width:1%">Range (Dev/Heavy/Light)</th>
+		<th style="width:0.3%">Temp. Disp.</th>
+		</tr>
+		"}
+	var/sensed_list = ""
+	for (var/datum/sensed_explosion/SE in sensed_explosions["z[z]"])
+		var/SE_dat = {"<tr>
+				<td style="text-align:center">[SE.time]</td>
+				<td style="text-align:center">[SE.area.name]</td>
+				<td style="text-align:center">([SE.x],[SE.y],[SE.z])</td>
+				"}
+		if (SE.cap)
+			SE_dat += {"<td style="text-align:center"><b>!![round(SE.cap*0.25)]  /  [round(SE.cap*0.5)]  /  [round(SE.cap)]!!</b></td>"}
+		else
+			SE_dat += {"<td style="text-align:center">[SE.dev] / [SE.heavy] / [SE.light]</td>"}
+		SE_dat += {"<td style="text-align:center">[SE.delay] sec</td></tr>"}
+		sensed_list = SE_dat + sensed_list
+	dat += sensed_list
+	return dat
+
+/obj/machinery/computer/bhangmeter/proc/checkPosition()
+	for(var/mob/M in watching_mobs)
+		if(get_dist(src,M) > 1)
+			stopWatching(M)
+
+/obj/machinery/computer/bhangmeter/proc/stopWatching(var/mob/user)
+	if(!user)
+		for(var/mob/M in watching_mobs)
+			if(M.client)
+				spawn(5)//we give it time to fade out
+					M.client.images -= watcher_maps["\ref[M]"]
+					qdel(watcher_maps["\ref[M]"])
+					M.client.screen -= watcher_buttons["\ref[M]"]
+				M.unregister_event(/event/face, src, /obj/machinery/station_map/proc/checkPosition)
+				animate(watcher_maps["\ref[M]"], alpha = 0, time = 5, easing = LINEAR_EASING)
+				if (watcher_buttons["\ref[M]"])
+					animate(watcher_buttons["\ref[M]"], alpha = 0, time = 5, easing = LINEAR_EASING)
+
+		watching_mobs = list()
+	else
+		if(user.client)
+			spawn(5)//we give it time to fade out
+				if(!(user in watching_mobs))
+					user.client.images -= watcher_maps["\ref[user]"]
+					qdel(watcher_maps["\ref[user]"])
+					watcher_maps -= "\ref[user]"
+					user.client.screen -= watcher_buttons["\ref[user]"]
+			user.unregister_event(/event/face, src, /obj/machinery/station_map/proc/checkPosition)
+			animate(watcher_maps["\ref[user]"], alpha = 0, time = 5, easing = LINEAR_EASING)
+			if (watcher_buttons["\ref[user]"])
+				animate(watcher_buttons["\ref[user]"], alpha = 0, time = 5, easing = LINEAR_EASING)
+
+			watching_mobs -= user
+
+/obj/machinery/computer/bhangmeter/proc/announce_explosion(var/datum/sensed_explosion/SE)
+	if(stat & (FORCEDISABLE|NOPOWER|BROKEN))
+		return
+	if (SE.z != original_zLevel)
+		return
+	if (last_announced_explosion && (world.time < (last_announcement + announcement_cooldown)))
+		var/new_score = SE.dev * 4 + SE.heavy * 2 + SE.light
+		var/old_score = last_announced_explosion.dev * 4 + last_announced_explosion.heavy * 2 + last_announced_explosion.light
+		if (old_score >= new_score)
+			return//We add a delay between announcements, unless the new explosion is larger than the last one.
```

Comment:
looks like you might be exiting before an intended potential speech behavior below, consider re-ordering

Reply:
well no, the point of that return IS to exit the proc without making any announcement.

If more than one explosion occurs within 5 seconds, only the first one gets announced, unless the second one is larger.